youtube_dl/extractor/sohu.py

```diff
@@ -77,6 +104,11 @@ def _fetch_data(vid_id, mytv=False):
                     % (format_id, i + 1, part_count))
 
                 part_info = part_str.split('|')
+
+                # Sanitize URL to prevent download failure
+                if part_info[0][-1] == '/' and su[i][0] == '/':
+                    su[i] = su[i][1:]
+
                 video_url = '%s%s?key=%s' % (part_info[0], su[i], part_info[3])
```

Comment:
It does not work. For example, when ```part_info[0]``` is ```http://70.39.188.10/sohu/s26h23eab6/2/``` and su[i] is ```/0/191/bymuey1CRhqd9706LeW3YD.mp4```, I need ```http://70.39.188.10/sohu/s26h23eab6/2/0/191/bymuey1CRhqd9706LeW3YD.mp4``` but I get ```http://70.39.188.10/0/191/bymuey1CRhqd9706LeW3YD.mp4```. The result is 404 not found.